### PR TITLE
update(cmake): Use TARGETS to install our libraries.

### PR DESCRIPTION
**Pull request by geraldcombs**
Use "install(TARGETS ..." instead of "install(FILES ..." to install
libsinsp and libscap. This is compatible with multi-config generators
such as Visual Studio. It also gives us control over where different
types of artifacts are installed.

Signed-off-by: Gerald Combs <gerald@wireshark.org>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

/area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

> /area libpman

> /area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

This fixes the "install" targets when using multi-config generators. Tested with Visual Studio, but should work with Xcode and Ninja Multi-Config.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmake/modules/libsinsp.cmake [hunk 1]
```diff
@@ -70,7 +70,10 @@ endif()
 
 add_subdirectory(${LIBSINSP_DIR}/userspace/libsinsp ${CMAKE_BINARY_DIR}/libsinsp)
 set(LIBSINSP_LIB "${PROJECT_BINARY_DIR}/libsinsp/libsinsp.a")
-install(FILES "${LIBSINSP_LIB}" DESTINATION "${CMAKE_INSTALL_LIBDIR}/${LIBS_PACKAGE_NAME}"
+install(TARGETS sinsp
+			ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}/${LIBS_PACKAGE_NAME}"
+			LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}/${LIBS_PACKAGE_NAME}"
+			RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
```
Comment (leogr) [inline cmake/modules/libsinsp.cmake:76]: cc @araujof @terylpt

**Review by terylt — COMMENTED** [review]
Looks okay to me. I think Cmake 2.8.5 should support TARGET - I found 2.8.12 documentation which does.   Is LIBRARY and RUNTIME there for completeness or does it actually install shared libs/binaries?

Nice work!  We can see whether we can test it in the next few days.  Bogged down with another issue right now.

**Comment by geraldcombs** [comment]
> Is LIBRARY and RUNTIME there for completeness or does it actually install shared libs/binaries?

They're there for completeness. RUNTIME in particular ensures that DLLs will be installed in the right place if we ever enable shared libraries.

**Comment by FedeDP** [comment]
I don't think we will make it for the 0.10.0. Moving to 0.11.0.
/milestone 0.11.0

**Comment by geraldcombs** [comment]
#933 includes these changes, so closing.

**Pull request closed without merging**